### PR TITLE
Fix poly shape regression - not being able to set an extrusion point.

### PR DESCRIPTION
**Pull request by claudiaantoun**
### Purpose of this PR

This PR fixes a regression in the poly shape tool, with the user not being able to set an extrusion point, that I caused when fixing these bugs (https://jira.unity3d.com/browse/PBLD-52 and https://jira.unity3d.com/browse/PBLD-70).

### Links

**Jira:** https://jira.unity3d.com/browse/PBLD-78

### CODE REVIEW

File: Runtime/Core/ProBuilderMeshFunction.cs [hunk 1]
```diff
@@ -368,6 +368,18 @@ public void ToMesh(MeshTopology preferredTopology = MeshTopology.Triangles)
...
+            {
+                EnsureMeshFilterIsAssigned();
+
+                if (usedInParticleSystem)
+                    MeshUtility.RestoreParticleSystem(this);
+
+                IncrementVersionIndex();
+
+                return;
+            }
```
Comment (modrimkus-unity) [inline Runtime/Core/ProBuilderMeshFunction.cs:381]: Would be nice if the code duplication could be avoided (i.e. compared to the very bottom code of this function). Also a comment might be nice for future changes that would explain why the materials/submesh code below can't be executed/is problematic when subMeshCount == 0.

**Review by modrimkus-unity — APPROVED** [review]
Approving but left a comment for discussion.

**Review by karl- — APPROVED** [review]
+1 to Mod's comment. Otherwise no issues.

File: Runtime/Core/ProBuilderMeshFunction.cs (hunk 1 above)
Comment (claudiaantoun) [reply Runtime/Core/ProBuilderMeshFunction.cs]: Good point, thanks for the feedback! Fixed in [4e34729](https://github.com/Unity-Technologies/com.unity.probuilder/pull/470/commits/4e34729e31f3f46b08484045802645828293c616)

**Pull request merged**